packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -927,111 +1004,162 @@ function commitLayoutEffectOnFiber(
-                  parentStateNode.effectDuration += effectDuration;
-                  break outer;
+            if (enableProfilerCommitHooks) {
+              if (typeof onCommit === 'function') {
+                onCommit(
+                  finishedWork.memoizedProps.id,
+                  phase,
+                  effectDuration,
+                  commitTime,
+                );
+              }
+
+              // Schedule a passive effect for this Profiler to call onPostCommit hooks.
+              // This effect should be scheduled even if there is no onPostCommit callback for this Profiler,
+              // because the effect is also where times bubble to parent Profilers.
+              enqueuePendingPassiveProfilerEffect(finishedWork);
+
+              // Propagate layout effect durations to the next nearest Profiler ancestor.
+              // Do not reset these values until the next render so DevTools has a chance to read them first.
+              let parentFiber = finishedWork.return;
+              outer: while (parentFiber !== null) {
+                switch (parentFiber.tag) {
+                  case HostRoot:
+                    const root = parentFiber.stateNode;
+                    root.effectDuration += effectDuration;
+                    break outer;
+                  case Profiler:
+                    const parentStateNode = parentFiber.stateNode;
+                    parentStateNode.effectDuration += effectDuration;
+                    break outer;
+                }
+                parentFiber = parentFiber.return;
               }
-              parentFiber = parentFiber.return;
             }
+          } catch (error) {
+            captureCommitPhaseError(finishedWork, finishedWork.return, error);
           }
         }
-        break;
       }
-      case SuspenseComponent: {
-        commitSuspenseHydrationCallbacks(finishedRoot, finishedWork);
-        break;
-      }
-      case SuspenseListComponent:
-      case IncompleteClassComponent:
-      case ScopeComponent:
-      case OffscreenComponent:
-      case LegacyHiddenComponent:
-      case TracingMarkerComponent: {
-        break;
+      break;
+    }
+    case SuspenseComponent: {
+      recursivelyTraverseLayoutEffects(
+        finishedRoot,
+        finishedWork,
+        committedLanes,
+      );
+      if (flags & Update) {
+        try {
+          commitSuspenseHydrationCallbacks(finishedRoot, finishedWork);
+        } catch (error) {
+          captureCommitPhaseError(finishedWork, finishedWork.return, error);
+        }
       }
-
-      default:
-        throw new Error(
-          'This unit of work tag should not have side-effects. This error is ' +
-            'likely caused by a bug in React. Please file an issue.',
-        );
+      break;
     }
-  }
+    case OffscreenComponent: {
+      const isModernRoot = (finishedWork.mode & ConcurrentMode) !== NoMode;
+      if (isModernRoot) {
```

Comment:
Do we typically inline this?

Reply:
What do you mean? Like as opposed to outside the switch statement? Generally I like to put these types of things as local as possible so that it's only checked in the branches that need it